engine/resources/docker.go

```diff
@@ -0,0 +1,415 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/purpleidea/mgmt/engine"
+	"github.com/purpleidea/mgmt/engine/traits"
+	"github.com/purpleidea/mgmt/util"
+)
+
+const (
+	// dockerClientVersion is the api version used by the client. See:
+	// https://docs.docker.com/develop/sdk/#api-version-matrix for more info.
+	dockerClientVersion = "v1.37"
+	// dockerContainerRunning is the running container state.
+	containerRunning = "running"
```

Comment:
I think the states should probably be public in case we end up pulling them in from the lang somehow, but I think we'll eventually need a clean solution for these types of constants. This is sort of nitpicky, but I mentioned it to get you thinking about what we might eventually do on the language side.